include/simdjson/common_defs.h

```diff
@@ -113,6 +116,11 @@ constexpr size_t DEFAULT_MAX_DEPTH = 1024;
     SIMDJSON_DISABLE_GCC_WARNING(-Wunused-variable)
   #define SIMDJSON_PRAGMA(P) _Pragma(#P)
   #define SIMDJSON_DISABLE_GCC_WARNING(WARNING) SIMDJSON_PRAGMA(GCC diagnostic ignored #WARNING)
+  #if defined(SIMDJSON_CLANG_VISUAL_STUDIO)
+  #define SIMDJSON_DISABLE_UNDESIRED_WARNINGS SIMDJSON_DISABLE_GCC_WARNING(-Wmicrosoft-include)
```

Comment:
I have seen this -Wmicrosoft-include thing before. What does it do?

Reply:
I actually figured you added it ... I just refactored it into this macro from its location in simdjson.h and simdjson.cpp. I'll check git blame in a bit.

Reply:
Looks like it came from @pps83's [ClangCL PR](https://github.com/simdjson/simdjson/pull/809). I didn't see a rationale for it there, but all I did was ctrl+F microsoft-include in that PR, so it's possible it's there in a deep reading :)

Reply:
Clang's explanation: "Diagnostic text: warning: #include resolved using non-portable Microsoft search rules as: *A*"

Reply:
[It sounds like](https://stackoverflow.com/questions/48538707/include-search-paths-from-included-header-varies-for-different-compilers) shows one way this manifests is MSVC searches the including file's directory as well as the includee for #include "file.h". It's probably worrying maybe, since we shouldn't be including things that aren't in -I or the current directory? But the fact that gcc works makes me wonder how it's even possible.